src/init.cpp

```diff
@@ -1812,7 +1812,8 @@ bool AppInit2()
 
     if (fMasterNode) {
         LogPrintf("IS MASTER NODE\n");
-        initMasternode(GetArg("-masternodeprivkey", ""), GetArg("-masternodeaddr", ""), true);
+        auto res = initMasternode(GetArg("-masternodeprivkey", ""), GetArg("-masternodeaddr", ""), true);
+        if (!res) UIError(*res.error);
```

Comment:
(07aec0cb4b1cd72f227b1d7626d7c69da0c6e9de)
Might be better to be sure that `res.error` is always initialized when`res.res` is false.
Easiest way is probably by just replacing the constructor `OperationResult(bool _res)` with a `OperationResult()`, that only sets `res` to `true`.
In other words, make it impossible to call `OperationResult(false)` (if the result is false, the error **must** be set).

Reply:
Or, even simpler... just make the optional error field private, and provide a guarded getter.
Added this in c3cf32326fb7d218ce3c1b2674e99ebf62076477.